### DISCUSSION TITLE
Fine-tuning FLAN-T5 to predict question rating

### DISCUSSION THREAD

**Discussion by ArturDev42** [McGill-NLP/feedbackqa#7] [Q&A] [answered]
@xhluca @sivareddyg 

**TL;DR:** Are there more comprehensive and distinct definitions of the ratings "Acceptable" and "Could be Improved" as described in the paper [https://arxiv.org/abs/2204.03025](url) ? Has anyone done a similar task as desribed above with this datatset?

Hello, for a university project I am fine-tuning a flan-t5 model [https://huggingface.co/docs/transformers/model_doc/flan-t5](url) on the FeedbackQA dataset for the classification task of predicting the rating of a question. 

My **input X** is structured as follows: Prompt + Question + Answer. Prompt explains what to do ("Rate the answer for the given question according to [...]"). The **label Y** is one of the two ratings provided in the datset.

However, after experimenting with different versions and training configurations of flan-t5 (small, base, large) the accruacy on the test set is very low, around 50%. Addtionally, it looks like the model has less issues with the categories "Excellent" and "Bad" than with the other two categories "Acceptable" and "Could be Improved".

<img width="454" alt="Bildschirm­foto 2023-07-16 um 20 17 36" src="https://github.com/McGill-NLP/feedbackqa/assets/46421604/8b411245-c267-4258-bd52-fb79fca810c9">

The distribution of ratings in the dataset as shown in the figure below shows that the classes "Excellent" and "Bad" are more often represented than the other two classes.

<img width="622" alt="Bildschirm­foto 2023-07-16 um 19 51 25" src="https://github.com/McGill-NLP/feedbackqa/assets/46421604/6289ffd5-46fd-4cd4-9089-e234c3d85250">

Looking into the paper [https://arxiv.org/abs/2204.03025](url) I could not really find a well-defined explanation for the different categories apart from the example shown in the figure below. To it seems as though the categories "Acceptable" and "Could be Improved" are not distinctly well-defined.

<img width="963" alt="Bildschirm­foto 2023-07-16 um 20 22 56" src="https://github.com/McGill-NLP/feedbackqa/assets/46421604/9ddf16d6-2b92-47e1-b99e-01393d74452f">

I would like to ask if there exists are more comprehensive and distinct definition for these categories? Has anyone done a similar task with this datatset?

Any answer would be very much appreciated, thanks a lot!

**Comment by xhluca** [comment, marked as answer]
Those 4 labels were the ones used during the data collection. There are no supplementary labels. You can review figure 2 for statistics of the different components. It is normal that the distinction between acceptable and could be improved are very subtle, as they are not as black & white as excellent or bad; I imagine if we used likert instead, 2-4 are harder to distinguish than 1 or 5.

If you have better labels, it'd be a great extension to the dataset. If you want to contribute, please create a PR on the huggingface dataset.
